feat: ProductDetail 컴포넌트 추가

eeoooo opened the pull request:
## 📑 상품상세 페이지 기본 구현

### 📅 PR 생성 날짜 
- 2022-09-04 

### 💬 작업 내용
- ProductDetail: 상품상세 페이지 메인이 되는 컴포넌트입니다. 기능 구현 없이 기초 구조 잡은 상태입니다. 
- ProductOverview: 상품상세 페이지 상단에 썸네일 이미지와 상품 기본 정보와 주문 탭(추후 분리 고려)으로 구성했습니다. 

### 메모
- 아직 넘겨받는 데이터 구성이 완전하지 않아, 만들어주신 데이터 기반으로 ProductDetail안에 변수로 만들어 추가 필요 데이터를 더해 사용 중입니다. 
- 메인 페이지에서 데이터 넘겨받으며 페이지 라우팅/네비게이팅을 하는 것에 대해 얘기가 덜 되어 해당 부분 구현 안 된 상태입니다. 
- P.S. Styled Component나 컴포넌트 구조 설계 등이 처음이라 코드가 정신 없습니다. 리뷰 많이 남겨주시면 겸손하게 많이 배워서 후딱 반영하겠습니다!

eeoooo committed:
feat: ProductDetail 컴포넌트 추가

eeoooo committed:
refactor: 불필요한 스타일 컴포넌트 정리

import문 가독성과 스타일링을 하지 않는 스타일 컴포넌트를 일반 HTML태그로 변경했습니다. 동시에 multlline import문의 구조를 보기 좋게 변경했습니다.

wjd2676 commented on src/components/ProductDetail/index.jsx:
```diff
@@ -0,0 +1,64 @@
+    return (
+        <Wrapper>
+            <ProductOverview data={data}/>
+            <Navigation>
```
Navigation 은 컴포넌트 분리가 필요해보입니다.
NavButton이 반복되서 사용되고있는데 상수데이터를 활용해보시는건 어떨까요?

eeoooo replied:
네 알겠습니다 :) 시도해보겠습니다!

wjd2676 commented on src/components/ProductDetail/index.jsx:
```diff
@@ -0,0 +1,64 @@
+                <NavButton onClick={onLinkClick}>리뷰</NavButton>
+                <NavButton onClick={onLinkClick}>문의</NavButton>
+            </Navigation>
+            <ProductFeature>
```
ProductFeature 태그도 컴포넌트 분리가 필요해보입니다!

wjd2676 commented on src/components/ProductOverview/style.js:
```diff
@@ -0,0 +1,101 @@
+    padding: 0.5rem;
+`;
+
+export const ShareProduct = styled(SaveProduct)`
```
SaveProduct의 스타일을 복사하고 아무런 Style이 추가되지 않는다면 ShareProduct 태그를 새로 만들필요가 있을지 다시 생각해봐야 할것같습니다.

wjd2676 commented on src/components/ProductOverview/style.js:
```diff
@@ -0,0 +1,101 @@
+    font-weight: 600;
+    border-radius: 5px;
+    border: 1px solid ${variables.colors.darkGreen};
+`;
```
선언되는 Styled Component 가 많다고 생각됩니다. Nesting을 활용해보시는게 좋아보입니다!

eeoooo replied:
넵 정리해뒀습니다! Nesting 엮어서 더 깔끔하게 시도해보겠습니다! 꼼꼼하게 멘션해주셔서 감사합니다 :-)

wjd2676 reviewed (commented):
수고 많으셨습니다!

ghbaekdev commented:
고생하셨습니다👍

CH4MD0M commented on src/components/ProductOverview/index.jsx:
```diff
@@ -0,0 +1,57 @@
+    Description, Price, Message, DeliveryInfo,SpanStrong, Span,
+    Order, Form, Label, DropDown, OrderButtons, Button } from './style';
+
+export const ProductOverview = ({ data }) => {
```
`data`에 포함되어 있는 부분을 디스트럭처링으로 받아오시면 더 좋을 것 같습니다.

eeoooo replied:
그럴 것 같네요! 코멘트 감사합니다! 수정하겠습니다 :-)

CH4MD0M reviewed (approved):
고생하셨습니다~

eeoooo committed:
Craco 실행 실패 이유로 npm업데이트 진행

eeoooo committed:
refactor: ProductFeature 컴포넌트 분리

PR피드백을 기반으로 ProductDetail에 있던 ProductFeature부분을 컴포넌트로 분리했습니다. 
ProductFeature안에는 Product에 대한 상세정보이미지, 리뷰 섹션, 문의 섹션이 들어가게 됩니다.

eeoooo committed:
rafactor: Deconstructing ProductOverview컴포넌트 props data

PR피드백 반영해서 ProductOverview에 받아온 props의 data를 deconstructing하여 각 자원을 이용하게 했습니다.
아예 받아오는 단에서 deconstructing하고 싶었지만 제대로 동작하지 않아 받아온 후에 분해하도록 하였습니다.

eeoooo committed:
refactor: ProductOrder 컴포넌트 분리

ProductOverview컴포넌트에서 상품 선택, 수량 추가, 총 가격 계산, 제출 버튼 내용을 ProductOrder컴포넌트로 분리했습니다.

eeoooo committed:
style: 상품상세 관련 컴포넌트들 return문 전에 공백 한 줄씩 추가

가독성 위해 추가하였습니다.

eeoooo committed:
feat: 제품 상세 option 추가, 삭제 구현

상품의 option 아이템 선택 시 렌더링하고, 삭제 가능하게 했습니다. 
수량에 따른 가격 업데이트 기능은 미구현 상태입니다.

jinux127 commented:
고생하셨습니다~!

스타일 컴포넌트를 매 요소마다 선언해주셨는데 이게 어떤 기능을 하는 컴포넌트인지 단순히 스타일이 적용된 컴포넌트인지 구분하기 힘들었어요. 

보통 스타일이 적용된 컴포넌트를 선언해줄때는 SC.NavButton 처럼 사용하거나 Styled를 붙여서 표현해주는걸 보긴 했던 것 같아요. 

찾아보니 이런 글이 있네요. 이해하시는데 도움이 될 것 같아서 [링크](https://velog.io/@sonic/%ED%9A%8C%EA%B3%A0-Styled-Components%EB%A5%BC-%EC%82%AC%EC%9A%A9%ED%95%98%EB%A9%B0)걸어드려요!  덕분에 저도 시야가 넓어졌어요 감사합니다~

eeoooo committed:
Merge branch 'main' into feature/detailPage

eeoooo committed:
feat: 리스트 아이템 클릭 시 상품상세 페이지 렌더링

경로에 클릭된 아이템 데이터를 담아 보내게 했습니다.
+
commit을 놓쳐서 상품상세페이지 option선택 기능에 대해 따로 커밋을 남기지 않았습니다. 해당 부분 업데이트도 포함되어 있습니다.

eeoooo committed:
feat: 주문 버튼 클릭 시 주문 페이지로 이동

버튼 클릭 시 oreder페이지로 이동하게 합니다.데이터가 간단하고, 단일 아이템별로 주문하기로 하였기에 useNavigate에 state를 넣어서 데이터를 전송하게 했습니다.

eeoooo committed:
Merge remote-tracking branch 'origin/main' into main

eeoooo committed:
Merge remote-tracking branch 'refs/remotes/origin/main'

Conflicts:
	package-lock.json
	src/App.js

eeoooo committed:
Merge branch 'main' of https://github.com/Wanted-Pre-Onboarding-FE-Team-12/wanted-pre-onboarding-fruitte

eeoooo committed:
Merge remote-tracking branch 'origin/main' into main
